### PR TITLE
CA-219436: Fix raw iscsi sr live-migrate

### PR DESCRIPTION
**Pull request by Annie-XIE**
VM with raw iSCSI SR cannot live migrate as there are more strict
checks on the supported operations, this fix will check config
item relax_xsm_sr_check, if it's true, check source and destination
SR's uuid, if they are the same, mark it as support live migrate.

Signed-off-by: Huan Xie <huan.xie@citrix.com>

### CODE REVIEW

File: ocaml/xapi/xapi_vm_migrate.ml [hunk 1]
```diff
@@ -113,10 +113,22 @@ let assert_sr_support_operations ~__context ~vdi_map ~remote ~ops =
     if not (List.for_all (fun op -> List.mem Smint.(string_of_capability op) sm_capabilities) ops) then
       raise (Api_errors.Server_error(Api_errors.sr_does_not_support_migration, [Ref.string_of sr]))
   in
+  let relaxed_xsm_sr_matches local_vdi_ref remote_sr_ref =
+    let source_sr_ref = Db.VDI.get_SR ~__context ~self:local_vdi_ref in
+    if !Xapi_globs.relax_xsm_sr_check then
+      begin
+       let source_sr_uuid = Db.SR.get_uuid ~__context ~self:source_sr_ref in
```
Comment (lindig) [inline ocaml/xapi/xapi_vm_migrate.ml]: Indentation is here 1 character but should be 2 characters.

**Comment by lindig** [comment]
Looks good to me. One thought: `relaxed_xsm_sr_matches` crams a lot into one identifier that becomes hard to read and I think it might be better to add a small comment on top about what is relaxed here and change it to a simpler `matching` or `is_matching`.

File: ocaml/xapi/xapi_vm_migrate.ml (hunk 1 above)
Comment (Annie-XIE) [inline ocaml/xapi/xapi_vm_migrate.ml]: Got it, thanks a lot, I will change to 2 characters for the code snippet

**Comment by Annie-XIE** [comment]
Sure, will re-work on that, make the func name easy to understand, thanks:)

**Comment by Annie-XIE** [comment]
@lindig I have modified with your comments and pushed again for review, could you please help have a review again? Thanks a lot.

**Comment by lindig** [comment]
Great. I'll briefly wait for other comments today and then merge it.

**Comment by Annie-XIE** [comment]
Thanks!

**Pull request merged**